### PR TITLE
Cluster map

### PR DESCRIPTION
**Pull request by rubyimrie**
Fixes the cluster map by using useLMarkerClusterCustom Adapted version of : {https://github.com/nuxt-modules/leaflet/blob/0ce672e60537210e733846dae6e40d9c75d0e5a3/src/runtime/composables/useLMarkerCluster.ts} which allows for passing through cluster options.

### CODE REVIEW

**Comment by Sephster** [comment]
It might be in my head but the animations look nicer in this than the previous version! Good job on solving this, I know it wasn't easy. Functionally works great.

File: components/filter-components/Timeline.vue [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 <script>
 import Slider from '@vueform/slider'
+import '@vueform/slider/themes/default.css'
```
Comment (Sephster) [inline components/filter-components/Timeline.vue:26]: I'm not certain but I don't think the slider is having any effect on the witches on the map. They aren't changing for me. Thanks Ruby. This might be expected and dealt with in another PR but wanted to flag here in case that isn't the case.

File: components/filter-components/Timeline.vue [hunk 2]
```diff
@@ -60,6 +61,7 @@ export default {
         this.parseDate(this.dates[newRange[1]].label)
       ];
       this.$emit("updatedRangeValue", dateRange);
+      console.log("Updated Date Range", dateRange);
```
Comment (Sephster) [inline components/filter-components/Timeline.vue:64]: Looks like a debugging statement. If so, we should remove this. Thank you

File: components/filter-components/Timeline.vue [hunk 3]
```diff
@@ -80,15 +82,22 @@ export default {
     },
     handleCustomInputRangeChange(newCustomInputRange) {
       this.numberRangeValue = newCustomInputRange;
+      console.log("New Number range value", newCustomInputRange);
```
Comment (Sephster) [inline components/filter-components/Timeline.vue:85]: Looks like a debugging statement. If so, we should remove this. Thank you

File: components/leaflet-maps/ClustersMap.vue [hunk 4]
```diff
@@ -142,26 +80,77 @@
      }
    },
    methods: {
+    fillMarkersArray(mapMarkers) {
+      this.markers = mapMarkers.map((markerData) => {
+        const lat = markerData.longLat[0];
+        const lng = markerData.longLat[1];
+
+        // Building the popup content as an HTML string
+        const popupContent = `
```
Comment (Sephster) [inline components/leaflet-maps/ClustersMap.vue:89]: Is this the best way to do this? I don't know Vue well enough to be certain but I would have thought we could render a custom vue component with all of this included for the popover and just pass the data in as props?

File: components/leaflet-maps/ClustersMap.vue [hunk 5]
```diff
@@ -142,26 +80,77 @@
...
+       this.$refs.myMap.leafletObject.flyTo(coords ,14);
      },
      emitMapData: function () {
        // Emmits an object containing the information about
        // where the center of the map is, the zoom, and what 
        // map type to change to when the map is turned off,
        // in this case changing to clustersOff.
-
-       let centerInfo = this.$refs.myMap.mapObject.getCenter();
+       console.log("Emitting Map Data")
```
Comment (Sephster) [inline components/leaflet-maps/ClustersMap.vue:144]: Looks like a debugging statement to remove.

File: components/leaflet-maps/ClustersMap.vue [hunk 6]
```diff
@@ -142,26 +80,77 @@
...
+       let centerInfo = this.$refs.myMap.leafletObject.getCenter();
+       console.log("Center Info", centerInfo)
```
Comment (Sephster) [inline components/leaflet-maps/ClustersMap.vue:146]: Looks like a debugging statement to remove

File: components/leaflet-maps/ClustersMap.vue [hunk 7]
```diff
@@ -142,26 +80,77 @@
...
        let centerArray = [centerInfo.lat, centerInfo.lng];
        let changeInfo = {
          center: centerArray,
-         zoom: this.$refs.myMap.mapObject.getZoom(),
+         zoom: this.$refs.myMap.leafletObject.getZoom(),
          changeTo: "clustersOff"
        };
+       console.log("Zoom", changeInfo.zoom)
```
Comment (Sephster) [inline components/leaflet-maps/ClustersMap.vue:153]: Looks like a debugging statement to remove

File: components/leaflet-maps/LeafletMapMain.vue [hunk 8]
```diff
@@ -17,7 +17,7 @@
           </label>
         </div>
           <!--Will chnage back to clusters when get clusters map working  -->
```
Comment (Sephster) [inline components/leaflet-maps/LeafletMapMain.vue:19]: Should this comment be removed now?

**Review by Sephster — CHANGES_REQUESTED** [review]
Minor changes requested but this is looking good. Thanks Ruby

File: components/filter-components/Timeline.vue (hunk 1 above)
Comment (Sephster) [reply components/filter-components/Timeline.vue]: I still don't think this is working @rubyimrie  - if you fix again, could you detail the fix and issue and leave it unresolved? Thank you

File: components/filter-components/Timeline.vue (hunk 2 above)
Comment (Sephster) [reply components/filter-components/Timeline.vue]: Thanks Ruby

File: components/filter-components/Timeline.vue (hunk 3 above)
Comment (Sephster) [reply components/filter-components/Timeline.vue]: Thanks Ruby

File: components/leaflet-maps/ClustersMap.vue (hunk 4 above)
Comment (Sephster) [reply components/leaflet-maps/ClustersMap.vue:89]: ahh no. That is cool to leave as is. Thanks for the explanation Ruby. Appreciate it

File: components/leaflet-maps/ClustersMap.vue (hunk 6 above)
Comment (Sephster) [reply components/leaflet-maps/ClustersMap.vue]: removed. Thanks Ruby

File: components/leaflet-maps/ClustersMap.vue (hunk 7 above)
Comment (Sephster) [reply components/leaflet-maps/ClustersMap.vue]: Removed. Thanks Ruby

File: components/leaflet-maps/LeafletMapMain.vue (hunk 8 above)
Comment (Sephster) [reply components/leaflet-maps/LeafletMapMain.vue]: Removed. Thank you

File: components/filter-components/Timeline.vue (hunk 1 above)
Comment (rubyimrie) [reply components/filter-components/Timeline.vue]: if (this.clustersOn) {
        this.clustersOn = !this.clustersOn;
      }
      
      have fixed, this bit of code i had commented out when the cluster map wasn't working and had forgotten to   change it back, when the timeline slider appears the map changes to normal map so that the changes are visible because with the clustering on the changes aren't noticeable

Comment (Sephster) [reply components/filter-components/Timeline.vue]: That's it working now. Thank you!

**Review by Sephster — APPROVED** [review]
All looks good. Happy to merge in. Thank you!
